crates/re_viewport/src/context_menu/actions.rs

```diff
@@ -544,3 +544,122 @@ fn create_space_view_for_selected_entities(
     ctx.viewport_blueprint
         .mark_user_interaction(ctx.viewer_context);
 }
+
+// ---
+
+/// Collapse or expand all items in the selection.
+// TODO(ab): the current implementation makes strong assumptions of which CollapseScope to use based
+// on the item type. This is brittle and will not scale if/when we add more trees to the UI. When
+// that happens, we will have to pass the scope to `context_menu_ui_for_item` and use it here.
+pub(super) enum CollapseExpandAllAction {
+    CollapseAll,
+    ExpandAll,
+}
+
+impl ContextMenuAction for CollapseExpandAllAction {
+    fn supports_selection(&self, ctx: &ContextMenuContext<'_>) -> bool {
+        // let's allow this action if at least one item supports it
+        ctx.selection
+            .iter()
+            .any(|(item, _)| self.supports_item(ctx, item))
+    }
+
+    fn supports_item(&self, ctx: &ContextMenuContext<'_>, item: &Item) -> bool {
+        // TODO(ab): in an ideal world, we'd check the fully expended/collapsed state of the item to
+        // avoid showing a command that wouldn't have an effect but that's lots of added complexity.
+        match item {
+            Item::StoreId(_) | Item::ComponentPath(_) => false,
+            Item::SpaceView(_) | Item::Container(_) | Item::InstancePath(_) => true,
```

Comment:
Well, containers _can_ be empty, but they always have the triangle icon (which is kinda weird, if you think of it, but I guess more natural given that they are _meant_ to contain something).

As for Instance paths, they are never empty, since they always have at least some components. When they show up in the blueprint tree, then they are `DataResult` and this case is properly handled.